policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -225,14 +230,69 @@ impl kubert::index::IndexNamespacedResource<Service> for Index {
 
     fn delete(&mut self, namespace: String, name: String) {
         tracing::debug!(name, namespace, "deleting service");
-        let service_ref = ServiceRef { name, namespace };
+        let service_ref = ResourceRef { name, namespace };
         self.service_info.remove(&service_ref);
         self.services_by_ip.retain(|_, v| *v != service_ref);
 
         self.reindex_services()
     }
 }
 
+impl kubert::index::IndexNamespacedResource<linkerd_k8s_api::UnmeshedNetwork> for Index {
+    fn apply(&mut self, u: linkerd_k8s_api::UnmeshedNetwork) {
+        let um: UnmeshedNetwork = u.into();
+        tracing::debug!(um.name, um.namespace, "indexing unmeshed network");
+
+        self.unmeshed_networks.insert(
+            ResourceRef {
+                name: um.name.clone(),
+                namespace: um.namespace.clone(),
+            },
+            um,
+        );
+    }
+
+    fn delete(&mut self, namespace: String, name: String) {
+        tracing::debug!(name, namespace, "deleting unmeshed networks");
+        let um_ref = ResourceRef { name, namespace };
+        self.unmeshed_networks.remove(&um_ref);
+    }
+}
+
+impl kubert::index::IndexNamespacedResource<Pod> for Index {
+    fn apply(&mut self, pod: Pod) {
+        let ns = pod.namespace().unwrap();
+        let name = pod.name_unchecked();
+        tracing::debug!(name, ns, "indexing pod");
+
+        if let Some(ips) = pod.status.and_then(|s| s.pod_ips) {
+            let pod_ref = ResourceRef {
+                name,
+                namespace: ns,
+            };
+            for ip in ips.iter() {
+                if let Some(ip) = &ip.ip {
+                    let pod_ip_addr = match IpAddr::from_str(ip) {
+                        Ok(addr) => addr,
+                        Err(error) => {
+                            tracing::error!(%error, "malformed PodIp");
+                            return;
+                        }
+                    };
+
+                    self.pods_by_ip.insert(pod_ip_addr, pod_ref.clone());
+                }
+            }
+        }
+    }
+
+    fn delete(&mut self, namespace: String, name: String) {
+        tracing::debug!(name, namespace, "deleting pod");
+        let pod_ref = ResourceRef { name, namespace };
+        self.pods_by_ip.retain(|_, v| *v != pod_ref);
+    }
```

Comment:
this won't remove the pod from the other mappings in this structure. this should, unless i'm mistaken, also be sure to remove entries in `unmeshed_networks`, `services_by_ip`, and `service_info`

Reply:
> this won't remove the pod from the other mappings in this structure. 

I am not sure I understand this comment. What other references are you referring to? The index impl here adds the pod to a map, when an apply is observed and then makes sure to remove it when we observe a delete event.What other references are there?

> this should, unless i'm mistaken, also be sure to remove entries in unmeshed_networks, services_by_ip, and service_info

 This bit of logic has nothing to do (at this point) with unmeshed_networks, services_by_ip or service_info structures. Separate code paths manage these. Am I missing something?

Reply:
this can be marked resolved, i was misunderstanding the contents of the index ✔️